### DISCUSSION TITLE
how to use vuex in vue?

### DISCUSSION THREAD

**Discussion by larrybrowndev** [drknzz/GitHub-Achievements#300] [Q&A] [answered]
I have a vue project and how can I use vuex there?

**Comment by topdev1228** [comment, marked as answer]
hi here is my solution:

import { createStore } from 'vuex';

const store = createStore({
  state() {
    return {
      count: 0,
    };
  },
  mutations: {
    increment(state) {
      state.count++;
    },
  },
  actions: {
    increment(context) {
      context.commit('increment');
    },
  },
});

export default store;

import { createApp } from 'vue';
import store from './store';
import App from './App.vue';

const app = createApp(App);
app.use(store);
app.mount('#app');


<template>
  <div>
    <p>Count: {{ count }}</p>
    <button @click="increment">Increment</button>
  </div>
</template>

<script>
import { defineComponent } from 'vue';
import { useStore } from 'vuex';

export default defineComponent({
  setup() {
    const store = useStore();

    return {
      count: store.state.count,
      increment() {
        store.dispatch('increment');
      },
    };
  },
});
</script>

**Reply by larrybrowndev** [reply]
thanks!
